SingularityUI/app/views/request.coffee

```diff
@@ -97,4 +106,84 @@ class RequestView extends View
+        @stopAutoTailPolling()
+
+        @listenTo @history, 'reset', @handleHistoryReset
+        @listenToOnce @activeTasks, 'add', @handleActiveTasksAdd
+
+        @autoTailPollInterval = interval 2000, =>
+            if @autoTailTaskFiles
+                @autoTailTaskFiles.fetch().error -> app.caughtError()  # we don't care about errors in this situation
+            else
+                @history.fetch()
+                @activeTasks.fetch()
+
+        @autoTailTimeout = timeout 60000, =>
+            @stopAutoTailPolling()
+            vex.close()
+            vex.dialog.alert
+                message: """
+                    <h3>Failure</h3>
+                    <code>#{ @autoTailFilename }</code> did not exist after #{ Math.floor(AUTO_TAIL_TIMEOUT / 60000) } minute(s).
+                """
+                buttons: [
+                    $.extend _.clone(vex.dialog.buttons.YES), text: 'OK'
+                ]
+
+    handleHistoryReset: (tasks) =>
+        timestamp = @autoTailTimestamp
+        matchingTask = tasks.find (task) -> task.get('taskId').startedAt > timestamp
+        if matchingTask
+            $('.auto-tail-checklist').addClass 'waiting-for-file'
+            @stopListening @activeTasks, 'add'
+            @autoTailTaskFiles = new TaskFiles [], taskId: matchingTask.get('id')
+
+    handleActiveTasksAdd: (task) =>
+        $('.auto-tail-checklist').addClass 'waiting-for-file'
+        @autoTailTaskId = task.get('id')
+        @autoTailTaskFiles = new TaskFiles [], taskId: @autoTailTaskId
+        @listenTo @autoTailTaskFiles, 'add', @handleTaskFilesAdd
+
+    handleTaskFilesAdd: =>
+        if @autoTailTaskFiles.findWhere({name: @autoTailFilename})
+            @stopAutoTailPolling()
+            @stopListening @history, 'reset', @handleHistoryReset
+            app.router.navigate "#task/#{@autoTailTaskId}/tail/#{@autoTailTaskId}/#{@autoTailFilename}", trigger: true
+            vex.close()
+
+    stopAutoTailPolling: ->
+        if @autoTailPollInterval
+            clearInterval @autoTailPollInterval
+        if @autoTailTimeout
+            clearTimeout @autoTailTimeout
+        @stopListening @activeTasks, 'add', @handleActiveTasksAdd
+        @stopListening @history, 'reset', @handleHistoryReset
+        if @autoTailTaskFiles
+            @stopListening @autoTailTaskFiles, 'add', @handleTaskFilesAdd
+        @autoTailTaskFiles = null
+
+    ## Prompt for cancelling the redirect after it's been initiated
+    showAutoTailWaitingDialog: ->
+        vex.dialog.alert
+            overlayClosesOnClick: false
+            message: """
+                <h3>Launching</h3>
+                <ol class="auto-tail-checklist">
+                    <li class="auto-tail-task-start">Waiting for task to launch</li>
+                    <li class="auto-tail-file-exists">Waiting for <code>#{ @autoTailFilename }</code> to exist</li>
+                </ol>
+                <div class='page-loader centered cushy'></div>
+            """
```

Comment:
I'm not sure if this is something you were intending on later or not but these should be in templates.